### DISCUSSION TITLE
Exploring Debugging Capabilities: A Closer Look at Debug Mode for srsRAN-4G's C++ Code

### DISCUSSION THREAD

**Discussion by ahmedelbushra** [srsran/srsRAN_4G#1274] [Q&A]
Is there a debug mode in srsRAN-4G that allows me to follow the C++ code execution in real-time while running the full srsRAN program on Linux? My goal is to gain a better understanding of how the physical layer within the C++ code is written and to identify the algorithms being used.

**Comment by Andrea99-beep** [comment]
Hi,
Have you find any solutions for this problem?

**Reply by ahmedelbushra** [reply]
Hi,
There is a debugging tool called gdbgui that allows you to debug and observe how an executable file works in real time. It also enables you to step through C code line by line by placing breakpoints and so on.
